api/filter_dimension_options.go

```diff
@@ -149,46 +162,84 @@ func (api *FilterAPI) getFilterBlueprintDimensionOptionHandler(w http.ResponseWr
 		return
 	}
 
+	b, err := json.Marshal(dimensionOption)
+	if err != nil {
+		log.ErrorCtx(r.Context(), errors.Wrap(err, "failed to marshal filter blueprint dimension option into bytes"), logData)
+		if auditErr := api.auditor.Record(r.Context(), getOptionAction, actionUnsuccessful, auditParams); auditErr != nil {
+			handleAuditingFailure(r.Context(), getOptionAction, actionUnsuccessful, w, auditErr, logData)
+			return
+		}
+		http.Error(w, internalError, http.StatusInternalServerError)
+		return
+	}
+
 	if auditErr := api.auditor.Record(r.Context(), getOptionAction, actionSuccessful, auditParams); auditErr != nil {
 		handleAuditingFailure(r.Context(), getOptionAction, actionSuccessful, w, auditErr, logData)
 		return
 	}
 
 	setJSONContentType(w)
-	w.WriteHeader(http.StatusNoContent)
+	w.WriteHeader(http.StatusOK)
```

Comment:
not a big issue, but this line is superfluous (if you've not already set status to something else), because the next line will trigger this behaviour, by default